department-of-veterans-affairs/caseflow
# Validate document ID when submitting decision

**Why**: Document IDs follow different formats depending on the work
product. We first introduced Document ID validation in #8575, but that
only applied to editing an existing Document ID. In this PR, we are
making sure Document IDs are validated during the attorney checkout
flow, when they are first created.

Resolves #8659

Testing Plan:
1. As an attorney, select an AMA case in your queue
2. From the Actions dropdown, select Decision ready for review
3. Complete the rest of the checkout flow until you reach the last page
where you can enter the Document ID
4. Enter an invalid ID, such as "123" and click Continue
Expected Result: You get a helpful error message stating what the
expected format is
5. Enter a valid ID, such as "12345-12345678" and click Continue
Expected Result: the case is successfully saved
5. Go back to your queue and select a Legacy case
6. From the Actions dropdown, select "Medical Request Ready for Review"
7. Select "OMO - VHA" or "OMO - IME"
4. Enter an invalid ID, such as "123" and click Continue
Expected Result: You get a helpful error message stating what the
expected format is
5. Enter a valid ID, such as "V1234567.123" (for VHA) and click Continue
Expected Result: the case is successfully saved
